code/game/machinery/buttons.dm

```diff
@@ -12,22 +12,59 @@
 	active_power_usage = 4
 	var/_wifi_id
 	var/datum/wifi/sender/wifi_sender
+	var/obj/item/device/assembly/trigger
 
 /obj/machinery/button/Initialize()
 	. = ..()
+	pixel_x = -DIR2PIXEL_X(dir)
+	pixel_y = -DIR2PIXEL_Y(dir)
 	update_icon()
 	if(_wifi_id && !wifi_sender)
 		wifi_sender = new/datum/wifi/sender/button(_wifi_id, src)
 
 /obj/machinery/button/Destroy()
 	qdel(wifi_sender)
 	wifi_sender = null
-	return..()
+	if(trigger)
+		trigger.forceMove(loc)
+	return ..()
 
 /obj/machinery/button/attack_ai(mob/user as mob)
 	return attack_hand(user)
 
-/obj/machinery/button/attackby(obj/item/weapon/W, mob/user as mob)
+/obj/machinery/button/dismantle()
+	playsound(loc, 'sound/items/Crowbar.ogg', 50, 1)
+	new/obj/item/frame/button(loc)

```

Comment:
Make a variable that's referenced here instead. Redefining the whole proc in this manner is silly for every subtype, instead of redefining a single variable.